classes/note/Note.php

```diff
@@ -3,173 +3,145 @@
+use Illuminate\Database\Eloquent\Builder;
+use Illuminate\Database\Eloquent\Casts\Attribute;
+use Illuminate\Database\Eloquent\Model;
+use PKP\db\DAO;
 
-class Note extends \PKP\core\DataObject
+class Note extends Model
 {
-    /**
-     * get user id of the note's author
-     *
-     * @return int
-     */
-    public function getUserId()
-    {
-        return $this->getData('userId');
-    }
+    use HasCamelCasing;
 
-    /**
-     * set user id of the note's author
-     *
-     * @param int $userId
-     */
-    public function setUserId($userId)
-    {
-        $this->setData('userId', $userId);
-    }
+    public const NOTE_ORDER_DATE_CREATED = 1;
+    public const NOTE_ORDER_ID = 2;
 
-    /**
-     * Return the user of the note's author.
-     *
-     * @return \PKP\user\User
-     */
-    public function getUser()
-    {
-        return Repo::user()->get($this->getUserId(), true);
-    }
+    protected $table = 'notes';
+    protected $primaryKey = 'note_id';
+    public $timestamps = false;
 
-    /**
-     * get date note was created
-     *
-     * @return string (YYYY-MM-DD HH:MM:SS)
-     */
-    public function getDateCreated()
-    {
-        return $this->getData('dateCreated');
-    }
+    protected $fillable = [
+        'assocType', 'assocId', 'userId',
+        'dateCreated', 'dateModified',
+        'title', 'contents'
+    ];
 
-    /**
-     * set date note was created
-     *
-     * @param string $dateCreated (YYYY-MM-DD HH:MM:SS)
-     */
-    public function setDateCreated($dateCreated)
+    protected function casts(): array
     {
-        $this->setData('dateCreated', $dateCreated);
+        return [
+            'assocType' => 'int',
+            'assocId' => 'int',
+            'userId' => 'int',
+            'dateCreated' => 'datetime',
+            'dateModified' => 'datetime'
+        ];
     }
 
     /**
-     * get date note was modified
-     *
-     * @return string (YYYY-MM-DD HH:MM:SS)
+     * Accessor and Mutator for primary key => id
      */
-    public function getDateModified()
+    protected function id(): Attribute
     {
-        return $this->getData('dateModified');
+        return Attribute::make(
+            get: fn($value, $attributes) => $attributes[$this->primaryKey] ?? null,
+            set: fn($value) => [$this->primaryKey => $value],
+        );
     }
 
     /**
-     * set date note was modified
+     * Return the user of the note's author.
      *
-     * @param string $dateModified (YYYY-MM-DD HH:MM:SS)
+     * @return \PKP\user\User
      */
-    public function setDateModified($dateModified)
+    public function getUser()
     {
-        $this->setData('dateModified', $dateModified);
+        return Repo::user()->get($this->userId, true);
     }
 
     /**
-     * get note contents
+     * Compatibility function for including note IDs in grids.
      *
-     * @return string
+     * @deprecated
      */
-    public function getContents()
+    public function getId()
```

Comment:
is this needed given that you can call the `id` accessor directly?

Reply:
It's necessary for some of the legacy code, e.g. GridHandler, which expects to deal with `DataObject` classes rather than Eloquent `Model`s. I had to add one for `Notification` models [here](https://github.com/pkp/pkp-lib/blob/4c31a124038bb79d0dfb4fdd0755f17c891c3d28/classes/notification/Notification.php#L135) too.

Reply:
Okay, thanks for clarifying